### PR TITLE
bug with createRootItem

### PR DESCRIPTION
**Pull request by arayi**
have traced through every single part of the registration chain. sometimes it inexplicably starts throwing errors like `"name" parameter must be a string` when I can't find anything in the registration chain that even USES a "name" parameter.

seems that createRootItem should be a similar function to `send` in `src/mail/registration_email.js` - any idea why it is not saving?

### CODE REVIEW

**Comment by jrob8577** [comment]
@arayi I'm confused as to why we're creating an entirely new branch and PR - the goal was to complete the feedback on the existing PR.  I'm still wondering how much we're seeing as broken by starting anew, and really don't want to invest the time in re-creating an existing branch and PR... The development and testing have already been done, so I feel like we're re-inventing the wheel with this approach?

**Comment by arayi** [comment]
My reasoning for the separate branch to do an incremental PR was that I wanted a second pair of eyes on this tiny part of the code as I fixed the merge conflict. All I did was roll it back and stage it in chunks in the appropriate files (the migration/model as a separate commit because it was a dependency and I knew for sure it was working, and part of routes/accounts.js needed to be in routes/accounts/register.js because of that refactor). Turns out I just needed a second pair of eyes, Jared helped me find a minor syntax error. The rest of the code had only minor conflicts so I'll have no trouble presenting a merge-ready PR first thing in the AM :zap:

Do you want me to do the final PR from Rachel's repo then?

**Comment by jrob8577** [comment]
I would like to minimize changes, and it seems to me that final PR from original branch serves that goal.

**Comment by arayi** [comment]
Okay, I'll have to get access to Rachel's repo, then, but since she is at lunch right now, here are the merge-ready changes for review :)

File: migrations/20161019214312-add-is-root-to-items.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+module.exports = {
+	up: function (queryInterface, Sequelize) {
```
Comment (jrob8577) [inline migrations/20161019214312-add-is-root-to-items.js]: Please use spaces instead of tabs.

File: config/config.js [hunk 2]
```diff
@@ -8,12 +8,15 @@ if( fs.existsSync( '.env' ) ) {
...
   },
   "test": {
-     "use_env_variable": "DATABASE_URL"
+     "use_env_variable": "DATABASE_URL",
+     "use_env_variable": "SITE_URL" 
   },
   "production": {
-     "use_env_variable": "DATABASE_URL"
+     "use_env_variable": "DATABASE_URL",
+     "use_env_variable": "SITE_URL"
```
Comment (jrob8577) [inline config/config.js]: Please revert these changes

File: routes/accounts/register.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 const encryptPassword = require( '../../auth/encryptPassword' )
 const RegistrationEmail = require( '../../src/mail/registration_email' )
+const { createRootItem } = require( '../items/item_response')
 
-const register = ( User, email, password ) =>
+const register = ( User, Item, email, password ) =>
   User.create({ email, password: encryptPassword( password ) })
-    .then( user => RegistrationEmail.send( user ) )
+  	.then( createRootItem(Item) )
+		.then( user => RegistrationEmail.send( user ) )
```
Comment (jrob8577) [inline routes/accounts/register.js]: Please use spaces instead of tabs on previous two lines, and ensure appropriate indentation. (`.then`s should be left aligned)

File: routes/items.js [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
 
-const { allItemsQuery, filteredItemsQuery, respondWithItems } = require( './items/item_response' )
-const { buildTree } = require( './items/tree_creation' )
+const { selectedItemsQuery, generateBreadcrumbs, respondWithItems } = require( './items/item_response' )
 const findAllItems = require('./items/find_all_items')
 
 router.get( '/', ( request, response ) => {
   const { Item } = request.app.get( 'models' )
-
+  const rootId = 0
```
Comment (jrob8577) [inline routes/items.js]: I don't see where this is used (or why it was added)?

File: routes/items/tree_creation.js [hunk 5]
```diff
@@ -2,10 +2,12 @@ const ItemTree = require( '../../src/item_tree' )
 
 const buildTree = items => ({ items, tree: new ItemTree( items )})
 
+const buildSubTree = item_id => ({items, tree}) =>
+	({items, tree: tree.subTree( item_id ) })
```
Comment (jrob8577) [inline routes/items/tree_creation.js]: Please replace tab with spaces

File: views/items/index.pug [hunk 6]
```diff
@@ -25,14 +25,33 @@ block content
...
+  each crumb in breadcrumbs.slice( 1, breadcrumbs.length - 1 )
+    li
+      a(href=`/items/${crumb.id}`)= crumb.title
+
   div
     h3 Your ToDo's:
 
-    .item-page
-      +display( tree, 0 )
+
```
Comment (jrob8577) [inline views/items/index.pug]: Remove extraneous whitespace - only one blank line is needed between logical blocks.

File: routes/items.js (hunk 4 above)
Comment (arayi) [inline routes/items.js]: it was in the original PR, looks like it's just extraneous so I'll rm

**Comment by arayi** [comment]
requested changes applied
